src/openms/source/ANALYSIS/MAPMATCHING/TransformationModelLinear.cpp

```diff
@@ -84,9 +84,22 @@ namespace OpenMS
         {
           points.push_back(Wm5::Vector2d(data_weighted[i].first, data_weighted[i].second));
         }
-        if (!Wm5::HeightLineFit2<double>(static_cast<int>(size), &points.front(), slope_, intercept_))
+        if (size == 2)
+        {
+          if (!Wm5::HeightLineFit2<double>(static_cast<int>(size), &points.front(), slope_, intercept_))
```

Comment:
why try fitting in the first place? Wouldn't computing the exact solution be enough?

Reply:
yes, you are right. I will do the exact calculation directly and add another exception in case the slope or intercept are nan or inf